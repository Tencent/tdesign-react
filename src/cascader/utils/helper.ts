import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
