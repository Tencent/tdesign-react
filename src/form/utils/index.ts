import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
