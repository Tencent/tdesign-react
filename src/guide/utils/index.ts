import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
