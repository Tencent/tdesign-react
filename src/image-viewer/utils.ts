export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
