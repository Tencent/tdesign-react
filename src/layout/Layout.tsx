import React, { useEffect, useState } from 'react';
import classNames from 'classnames';
import useConfig from '../hooks/useConfig';
import { StyledProps } from '../common';
import { TdLayoutProps, TdHeaderProps, TdFooterProps } from './type';
import Aside from './Aside';

export interface LayoutProps extends TdLayoutProps, StyledProps, React.HTMLAttributes<HTMLDivElement> {
  children?: React.ReactNode;
}
export interface HeaderProps extends TdHeaderProps, StyledProps, React.HTMLAttributes<HTMLElement> {
  children?: React.ReactNode;
}
export interface ContentProps extends StyledProps, React.HTMLAttributes<HTMLElement> {
  children?: React.ReactNode;
}
export interface FooterProps extends TdFooterProps, StyledProps, React.HTMLAttributes<HTMLElement> {
  children?: React.ReactNode;
}

const Header = (props: HeaderProps) => {
  const { classPrefix } = useConfig();
  const { className, style = {}, children, height, ...others } = props;
  const renderHeight = isNaN(Number(height)) ? height : `${height}px`;

  const headerClassNames = classNames(`${classPrefix}-layout__header`, className);
  return (
    <header
      className={headerClassNames}
      style={{
        height: renderHeight,
        ...style,
      }}
      {...others}
    >
      {children}
    </header>
  );
};

const Footer = (props: FooterProps) => {
  const { classPrefix } = useConfig();
  const { className, style = {}, children, height, ...others } = props;
  const renderHeight = isNaN(Number(height)) ? height : `${height}px`;
  const footerClassNames = classNames(`${classPrefix}-layout__footer`, className);
  return (
    <footer
      className={footerClassNames}
      style={{
        height: renderHeight,
        ...style,
      }}
      {...others}
    >
      {children}
    </footer>
  );
};

const Content = (props: ContentProps) => {
  const { classPrefix } = useConfig();
  const { className, style, children, ...others } = props;
  const contentClassNames = classNames(`${classPrefix}-layout__content`, className);
  return (
    <main className={contentClassNames} style={style} {...others}>
      {children}
    </main>
  );
};

/**
 * 布局组件
 */
const Layout = (props: LayoutProps) => {
  const { direction, className, style, children, ...otherLayoutProps } = props;
  const [asides, setAsides] = useState([]);

  useEffect(() => {
    React.Children.forEach(children, (child: React.ReactChild) => {
      if (!child || typeof child !== 'object') return;

      if (child.type === Aside) setAsides([child]);
    });
  }, [children]);

  const { classPrefix } = useConfig();
  const layoutClassNames = classNames(
    `${classPrefix}-layout`,
    {
      [`${classPrefix}-layout--with-sider`]: !!asides.length,
      [`${classPrefix}-layout__direction-${direction}`]: direction,
    },
    className,
  );

  return (
    <div className={layoutClassNames} style={style} {...otherLayoutProps}>
      {children}
    </div>
  );
};

Layout.Header = Header;
Layout.Content = Content;
Layout.Footer = Footer;
Layout.Aside = Aside;

Header.displayName = 'Header';
Content.displayName = 'Content';
Footer.displayName = 'Footer';
Layout.displayName = 'Layout';

export default Layout;
