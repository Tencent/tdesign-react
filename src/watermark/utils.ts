const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
