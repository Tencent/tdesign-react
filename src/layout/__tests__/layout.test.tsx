import { render } from '@test/utils';
import React from 'react';
import Layout from '../index';

const { Header, Content, Footer, Aside } = Layout;

// TODO
describe('Layout 组件测试', () => {
  test('Layout direction ', () => {
    const { container } = render(<Layout direction="horizontal" />);
    const layoutElement = container.querySelector('.t-layout');
    expect(layoutElement).not.toBeNull();
    const horizontal = container.querySelector('.t-layout__direction-horizontal');
    expect(horizontal).not.toBeNull();
  });

  test('Header ', () => {
    const { container } = render(<Header />);
    expect(container).not.toBeNull();
  });

  test('Content ', () => {
    const { container } = render(<Content />);
    expect(container).not.toBeNull();
  });

  test('Footer ', () => {
    const { container } = render(<Footer />);
    expect(container).not.toBeNull();
  });

  test('Aside ', () => {
    const { container } = render(<Aside />);
    expect(container).not.toBeNull();
  });
});
