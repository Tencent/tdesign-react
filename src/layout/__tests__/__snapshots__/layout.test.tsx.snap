// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aside.jsx 1`] = `
<DocumentFragment>
  <h4>
    侧边导航布局
  </h4>
  <div
    class="t-layout t-layout--with-sider"
  >
    <aside
      class="t-layout__sider"
      style="width: 232px; max-width: 232px; min-width: 232px; flex: 0 0 232px;"
    >
      <div
        class="t-default-menu"
        style="width: 100%; height: 100%; box-shadow: none;"
      >
        <div
          class="t-default-menu__inner"
        >
          <div
            class="t-menu__logo"
          >
            <img
              alt="logo"
              src="https://www.tencent.com/img/index/menu_logo_hover.png"
              width="136"
            />
          </div>
          <ul
            class="t-menu t-menu--scroll"
          >
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                侧边内容一
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                侧边内容二
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                侧边内容三
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                侧边内容四
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                侧边内容无
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <div
      class="t-layout"
    >
      <main
        class="t-layout__content"
      >
        <div>
          Content
        </div>
      </main>
      <footer
        class="t-layout__footer"
      >
        Copyright @ 2019-2020 Tencent. All Rights Reserved
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <h4>
    顶部导航布局
  </h4>
  <div
    class="t-layout"
  >
    <header
      class="t-layout__header"
    >
      Header
    </header>
    <main
      class="t-layout__content"
    >
      Content
    </main>
    <footer
      class="t-layout__footer"
    >
      Copyright @ 2019-2021 Tencent. All Rights Reserved
    </footer>
  </div>
  <br />
  <h4>
    侧边导航布局
  </h4>
  <div
    class="t-layout t-layout--with-sider"
  >
    <aside
      class="t-layout__sider"
      style="width: 232px; max-width: 232px; min-width: 232px; flex: 0 0 232px;"
    >
      Aside
    </aside>
    <div
      class="t-layout"
    >
      <main
        class="t-layout__content"
      >
        Content
      </main>
      <footer
        class="t-layout__footer"
      >
        Copyright @ 2019-2021 Tencent. All Rights Reserved
      </footer>
    </div>
  </div>
  <br />
  <h4>
    组合导航布局
  </h4>
  <div
    class="t-layout"
  >
    <header
      class="t-layout__header"
    >
      Header
    </header>
    <div
      class="t-layout t-layout--with-sider"
    >
      <aside
        class="t-layout__sider"
        style="width: 232px; max-width: 232px; min-width: 232px; flex: 0 0 232px;"
      >
        Aside
      </aside>
      <div
        class="t-layout"
      >
        <main
          class="t-layout__content"
        >
          Content
        </main>
        <footer
          class="t-layout__footer"
        >
          Copyright @ 2019-2021 Tencent. All Rights Reserved
        </footer>
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-layout"
  >
    <header
      class="t-layout__header"
    >
      Header
    </header>
    <div
      class="t-layout t-layout--with-sider"
    >
      <div
        class="t-layout"
      >
        <main
          class="t-layout__content"
        >
          Content
        </main>
        <footer
          class="t-layout__footer"
        >
          Copyright @ 2019-2021 Tencent. All Rights Reserved
        </footer>
      </div>
      <aside
        class="t-layout__sider"
        style="width: 232px; max-width: 232px; min-width: 232px; flex: 0 0 232px;"
      >
        Aside
      </aside>
    </div>
  </div>
  <br />
  <div
    class="t-layout"
  >
    <header
      class="t-layout__header"
    >
      Header
    </header>
    <div
      class="t-layout t-layout--with-sider"
    >
      <aside
        class="t-layout__sider"
        style="width: 80px; max-width: 80px; min-width: 80px; flex: 0 0 80px;"
      >
        Aside
      </aside>
      <div
        class="t-layout"
      >
        <main
          class="t-layout__content"
        >
          Content
        </main>
        <footer
          class="t-layout__footer"
        >
          Copyright @ 2019-2021 Tencent. All Rights Reserved
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`combine.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-item--layout"
  >
    <div
      class="t-layout"
    >
      <header
        class="t-layout__header"
      >
        <div
          class="t-head-menu t-menu--light"
        >
          <div
            class="t-head-menu__inner"
          >
            <div
              class="t-menu__logo"
            >
              <img
                alt="logo"
                src="https://www.tencent.com/img/index/menu_logo_hover.png"
                width="136"
              />
            </div>
            <ul
              class="t-menu"
            >
              <li
                class="t-menu__item t-is-active t-menu__item--plain"
              >
                <span
                  class="t-menu__content"
                >
                  已选内容
                </span>
              </li>
              <li
                class="t-menu__item t-menu__item--plain"
              >
                <span
                  class="t-menu__content"
                >
                  菜单内容一
                </span>
              </li>
              <li
                class="t-menu__item t-menu__item--plain"
              >
                <span
                  class="t-menu__content"
                >
                  菜单内容二
                </span>
              </li>
              <li
                class="t-menu__item t-is-disabled t-menu__item--plain"
              >
                <span
                  class="t-menu__content"
                >
                  菜单内容三
                </span>
              </li>
            </ul>
            <div
              class="t-menu__operations"
            >
              <div
                class="t-menu__operations"
              >
                <svg
                  class="t-icon t-icon-search t-menu__operations-icon"
                  fill="none"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                >
                  <path
                    d="M9.51 10.22a4.76 4.76 0 11.7-.7l3.54 3.52-.7.71-3.54-3.53zm.77-3.7a3.76 3.76 0 10-7.53 0 3.76 3.76 0 007.53 0z"
                    fill="currentColor"
                    fill-opacity="0.9"
                  />
                </svg>
                <svg
                  class="t-icon t-icon-notification-filled t-menu__operations-icon"
                  fill="none"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                >
                  <path
                    d="M8.5 2.03V1h-1v1.03a4.5 4.5 0 00-4 4.47V11l-.9 1.2a.5.5 0 00.4.8h2.55a2.5 2.5 0 004.9 0H13a.5.5 0 00.4-.8l-.9-1.2V6.5a4.5 4.5 0 00-4-4.47z"
                    fill="currentColor"
                    fill-opacity="0.9"
                  />
                </svg>
                <svg
                  class="t-icon t-icon-home t-menu__operations-icon"
                  fill="none"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                >
                  <path
                    d="M6 11v1h4v-1H6z"
                    fill="currentColor"
                    fill-opacity="0.9"
                  />
                  <path
                    d="M7.65 1.65c.2-.2.5-.2.7 0l6.5 6.5-.7.7L13 7.71v5.79a1 1 0 01-1 1H4a1 1 0 01-1-1V7.7L1.85 8.86l-.7-.7 6.5-6.5zM8 2.7l-4 4v6.79h8V6.7l-4-4z"
                    fill="currentColor"
                    fill-opacity="0.9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="t-layout t-layout--with-sider"
      >
        <aside
          class="t-layout__sider"
          style="width: 232px; max-width: 232px; min-width: 232px; flex: 0 0 232px;"
        >
          <div
            class="t-default-menu"
            style="width: 232px; margin-right: 50px; height: 550px;"
          >
            <div
              class="t-default-menu__inner"
            >
              <ul
                class="t-menu t-menu--scroll"
              >
                <li
                  class="t-menu__item t-is-active"
                >
                  <svg
                    class="t-icon t-icon-dashboard"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <g
                      fill="currentColor"
                      fill-opacity="0.9"
                      opacity="0.9"
                    >
                      <path
                        d="M13.87 4.9l-4.5 4.5-.7-.7 4.5-4.5.7.7z"
                      />
                      <path
                        d="M8.02 11.05a1 1 0 110-2 1 1 0 010 2zm0 1a2 2 0 100-4 2 2 0 000 4z"
                      />
                      <path
                        d="M8 3.5a6.5 6.5 0 00-5.34 10.21l-.82.58a7.5 7.5 0 019.63-10.93l-.46.88A6.47 6.47 0 008 3.5zm6.5 6.5c0-1.08-.26-2.1-.73-3l.88-.46a7.47 7.47 0 01-.5 7.75l-.81-.58A6.47 6.47 0 0014.5 10z"
                      />
                    </g>
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    仪表盘
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-server"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M7 5.25H4v-1h3v1z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                    <path
                      d="M1.5 2.5c0-.28.22-.5.5-.5h12c.28 0 .5.22.5.5V7a.5.5 0 01-.5.5H2a.5.5 0 01-.5-.5V2.5zm1 4h11V3h-11v3.5zM4 11.75h3v-1H4v1z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                    <path
                      d="M1.5 9c0-.28.23-.5.5-.5h12c.28 0 .5.22.5.5v4.5a.5.5 0 01-.5.5H2a.5.5 0 01-.5-.5V9zm1 4h11V9.5h-11V13z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    资源列表
                  </span>
                </li>
                <li
                  class="t-menu__item t-menu__item--plain"
                >
                  <span
                    class="t-menu__content"
                  >
                    <svg
                      class="t-icon t-icon-root-list"
                      fill="none"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                    >
                      <g
                        fill="currentColor"
                        fill-opacity="0.9"
                        opacity="0.9"
                      >
                        <path
                          d="M4.5 5h7v1h-7V5zM9 7.5H4.5v1H9v-1z"
                        />
                        <path
                          d="M3 2a1 1 0 00-1 1v10a1 1 0 001 1h10a1 1 0 001-1V3a1 1 0 00-1-1H3zm10 1v10H3V3h10z"
                        />
                      </g>
                    </svg>
                    根目录
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-control-platform"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M14.5 4.25L8.47.77a.94.94 0 00-.94 0L1.5 4.25v6.96c0 .33.18.64.47.81L8 15.5l6.03-3.48a.94.94 0 00.47-.81V4.25zM8 7.42L3 4.54l5-2.89 5 2.89-5 2.88zm.5.87l5-2.89v5.77l-5 2.89V8.29zm-1 0v5.77l-5-2.89V5.4l5 2.89z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    调度平台
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-precise-monitor"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M7.5 1.5V5h1V1.5h-1zM6.23 5.53L2.7 1.99l-.71.7 3.54 3.54.7-.7zm3.54 0l3.53-3.54.71.7-3.54 3.54-.7-.7zM9 8a1 1 0 01-1.87.5H1.5v-1h5.63A1 1 0 019 8zM5.53 9.77L1.99 13.3l.7.71 3.54-3.54-.7-.7zm4.94 0l3.54 3.53-.7.71-3.54-3.54.7-.7zM14.5 7.5H11v1h3.5v-1zm-7 7V11h1v3.5h-1z"
                      fill="currentColor"
                      opacity="0.9"
                    />
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    精准监控
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-mail"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M1.5 4a1 1 0 011-1h11a1 1 0 011 1v8a1 1 0 01-1 1h-11a1 1 0 01-1-1V4zm11.6 0H2.9L8 7.4 13.1 4zm-10.6.93V12h11V4.93L8 8.6 2.5 4.93z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    消息区
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-user-circle"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <g
                      fill="currentColor"
                      opacity="0.9"
                    >
                      <path
                        d="M8 9a2.78 2.78 0 100-5.56A2.78 2.78 0 008 9zm0-1a1.78 1.78 0 110-3.56A1.78 1.78 0 018 8z"
                      />
                      <path
                        d="M8 15A7 7 0 108 1a7 7 0 000 14zm5-3.68A10.2 10.2 0 008 10c-1.79 0-3.47.48-5 1.32a6 6 0 1110 0zm-.64.8a5.98 5.98 0 01-8.72 0 9.17 9.17 0 018.72 0z"
                      />
                    </g>
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    个人中心
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-play-circle"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <g
                      fill="currentColor"
                      fill-opacity="0.9"
                      opacity="0.9"
                    >
                      <path
                        d="M11.55 7.74c.2.12.2.4 0 .52l-5.1 2.94a.3.3 0 01-.45-.26V5.06a.3.3 0 01.45-.26l5.1 2.94z"
                      />
                      <path
                        d="M8 1a7 7 0 110 14A7 7 0 018 1zm0 1a6 6 0 100 12A6 6 0 008 2z"
                      />
                    </g>
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    视频区
                  </span>
                </li>
                <li
                  class="t-menu__item"
                >
                  <svg
                    class="t-icon t-icon-edit-1"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <g
                      fill="currentColor"
                      fill-opacity="0.9"
                      opacity="0.9"
                    >
                      <path
                        d="M14.13 4.95L10.9 1.71l.7-.71 3.25 3.24-.7.71zM5.97 13.11l-3.61.72a.3.3 0 01-.35-.35l.72-3.61 7.3-7.3 3.24 3.24-7.3 7.3zm5.89-7.3l-1.83-1.83-6.38 6.38-.46 2.29 2.29-.46 6.38-6.38zM15 11h-4v1h4v-1zM15 13H8.5v1H15v-1z"
                      />
                    </g>
                  </svg>
                  <span
                    class="t-menu__content"
                  >
                    资源编辑
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </aside>
        <div
          class="t-layout"
        >
          <main
            class="t-layout__content"
          >
            <div>
              Content
            </div>
          </main>
          <footer
            class="t-layout__footer"
          >
            Copyright @ 2019-2020 Tencent. All Rights Reserved
          </footer>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`top.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-layout"
  >
    <header
      class="t-layout__header"
    >
      <div
        class="t-head-menu t-menu--light"
      >
        <div
          class="t-head-menu__inner"
        >
          <div
            class="t-menu__logo"
          >
            <img
              alt="logo"
              src="https://www.tencent.com/img/index/menu_logo_hover.png"
              width="136"
            />
          </div>
          <ul
            class="t-menu"
          >
            <li
              class="t-menu__item t-is-active t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                已选内容
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                菜单内容一
              </span>
            </li>
            <li
              class="t-menu__item t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                菜单内容二
              </span>
            </li>
            <li
              class="t-menu__item t-is-disabled t-menu__item--plain"
            >
              <span
                class="t-menu__content"
              >
                菜单内容三
              </span>
            </li>
          </ul>
          <div
            class="t-menu__operations"
          >
            <div
              class="t-menu__operations"
            >
              <svg
                class="t-icon t-icon-search t-menu__operations-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M9.51 10.22a4.76 4.76 0 11.7-.7l3.54 3.52-.7.71-3.54-3.53zm.77-3.7a3.76 3.76 0 10-7.53 0 3.76 3.76 0 007.53 0z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <svg
                class="t-icon t-icon-notification-filled t-menu__operations-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8.5 2.03V1h-1v1.03a4.5 4.5 0 00-4 4.47V11l-.9 1.2a.5.5 0 00.4.8h2.55a2.5 2.5 0 004.9 0H13a.5.5 0 00.4-.8l-.9-1.2V6.5a4.5 4.5 0 00-4-4.47z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <svg
                class="t-icon t-icon-home t-menu__operations-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6 11v1h4v-1H6z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M7.65 1.65c.2-.2.5-.2.7 0l6.5 6.5-.7.7L13 7.71v5.79a1 1 0 01-1 1H4a1 1 0 01-1-1V7.7L1.85 8.86l-.7-.7 6.5-6.5zM8 2.7l-4 4v6.79h8V6.7l-4-4z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="t-layout__content"
    >
      <div>
        Content
      </div>
    </main>
    <footer
      class="t-layout__footer"
    >
      Copyright @ 2019-2020 Tencent. All Rights Reserved
    </footer>
  </div>
</DocumentFragment>
`;
